### ISSUE TITLE
Pattern match rules not strict enough

### ISSUE THREAD

**Issue by lauriii** [grasmash/drupal-issue-chrome#1]
There are URLs in Drupal.org that lead to pages that are not nodes. For example, https://www.drupal.org/pift-ci-job/[CI-JOB-ID] is a valid URL pattern. If there's a link to a page with that URL pattern, this browser plugin tries to process it and shows a [Invalid NID] before the link. 

### Proposed solution
Make the pattern match rules stricter so that it doesn't assume that all Drupal.org URLs ending with a numeric ID leads into a node.

**Comment by wimleers** [comment]
Yep, ran into this too with project links, like `https://www.drupal.org/project/entity_embed`.

Worse: as soon as this extension chokes on one of those unsupported URLs, none of the subsequent URLs are processed anymore. Not even after minutes of waiting. Timing out after some time would fix this. It seems this is actually failing because `processAnchorElement()` only handles `readyState === 4`, i.e. it only handles successful responses.

**Comment by wimleers** [comment]
Oh and even worse, look at https://www.drupal.org/project/drupal/issues/2847678#comment-13119540 for example. 

That comment links to `https://www.drupal.org/docs/8/testing/converting-d7-simpletests-to-drupal-8`, yet it shows it as `#8: Let users cancel their accounts`, i.e. the extension thinks it points to `https://drupal.org/node/8`!

**Comment by wimleers** [comment]
I'm disabling this browser extension for now because that last point can lead to me completely misinterpreting what I'm reading 😨

Still, very interesting project overall :) Looking forward to having these edge cases ironed out!

**Comment by TravisCarden** [comment]
Another failing test case patch files. Example:

* **Before:** `https://www.drupal.org/files/issues/2019-03-15/2241865-63.patch`
* **After:** `<a href="https://www.drupal.org/files/issues/2019-03-15/2241865-63.patch" class="external-link" rel="nofollow" title="closed (fixed)"><span class="drupalorg-issue issue-status-7">#2019: Online.module</span></a>`

**Comment by grasmash** [comment]
I’ve resolved a number of issues with pattern matching. Most have been resolved in 0.0.4, which is available now. An additional fix will be in 0.0.5, which is in queue for publication now. Should be available in a day or so. Please try 0.0.5 and let me know!

**Comment by grasmash** [comment]
Test cases:
```
// Pattern should match:
// https://www.drupal.org/project/drupal/issues/2982684
// https://www.drupal.org/project/composer_initiative/issues/3053800
// Pattern should not match:
// https://www.drupal.org/docs/8/modules/workspace
// https://www.drupal.org/project/entity_embed
// https://www.drupal.org/project/ctools/releases/8.x-3.2
// https://www.drupal.org/files/issues/2019-03-15/2241865-63.patch
```
See https://regex101.com/r/zTrAow/1

**Comment by grasmash** [comment]
Closing. Fixed all referenced examples in 0.0.5.
